tests/components/openweathermap/test_config_flow.py

```diff
@@ -28,53 +23,55 @@
     CONF_API_KEY: "foo",
     CONF_LATITUDE: 50,
     CONF_LONGITUDE: 40,
-    CONF_MODE: DEFAULT_FORECAST_MODE,
     CONF_LANGUAGE: DEFAULT_LANGUAGE,
 }
 
 VALID_YAML_CONFIG = {CONF_API_KEY: "foo"}
 
 
-async def test_form(hass: HomeAssistant) -> None:
-    """Test that the form is served with valid input."""
-    mocked_owm = _create_mocked_owm(True)
+def _create_mocked_owm_client(is_valid: bool):
+    mocked_owm_client = MagicMock()
+    mocked_owm_client.validate_key = AsyncMock(return_value=is_valid)
+    mocked_owm_client.get_weather = AsyncMock(return_value=None)
 
-    with patch(
-        "pyowm.weatherapi25.weather_manager.WeatherManager",
-        return_value=mocked_owm,
-    ):
-        result = await hass.config_entries.flow.async_init(
-            DOMAIN, context={"source": SOURCE_USER}
-        )
+    return mocked_owm_client
 
-        assert result["type"] is FlowResultType.FORM
-        assert result["step_id"] == "user"
-        assert result["errors"] == {}
 
-        result = await hass.config_entries.flow.async_init(
-            DOMAIN, context={"source": SOURCE_USER}, data=CONFIG
-        )
+@patch("pyopenweathermap.OWMClient", return_value=_create_mocked_owm_client(True))
```

Comment:
Usually we patch at the point of import instead of globally

Reply:
could you, please, show me an example how it should be done?

Reply:
done